IBMa/equal-access
# sync dev branch

* Update for Puppeteer 19

* Update imports for ESM

* Fix object verification

* Fix ACT runs

* Throw out bounds differences within 2

* Update cypress package-lock

* Split cypress test

* Convert to absolute url for Windows

* Update gulpfile

* Split into cjs and mjs

* Use modules for testing

* Tests to mjs

* Fix extension

* More file extension changes

* Fix require check

* Fix file extension

* Update file extensions

* Fix testcase to work on Windows

* Update meta hack for ACT test

* chore(node,karma): Update selenium-webdriver (4.9.2) and add more error logs (#1414)

* update package.json
* update web driver version and fix related errors #1356
* replace setDefaultService function in chrome #1356
* add more exception handling #1356
* more changes on the errors #1356
* clean up the code #1356
* add more error logs #1356
* update the boilerplates selenium-driver version from 4.3.0 to 4.8.0 #1356
* Use latest Selenium ---------



* chore(cypress): Update to latest Cypress (12.12.0)  (#1451)

* Update files to Cypress 12
* Use Node 18 for cypress
* Update boilerplate
* Put boilerplate dependency back
* Update baseline
* Run tests on Node 18
* Update test server packages
* Update lock file
* Try update ubuntu
* Split server and run commands
* Wrong command

* Selenium refix

* Wrong order

* Add jest boilerplate, fix terser typeof issue (#1461)

* Build failure

* import problems

* merge

* Issues with imports

* Import problems

* Fix module imports

* import problems

* Deal with cjs vs mjs for __dirname

* Fix filename

* Don't support 14.x with the 4.x branch

* Fix merge conflicts

* update the puppeteer version

* update the file

* update the package.json files

* update the comment

* update configuration files

* change from require to import

* change require() to import

* clean the code

* update the config files

* update the config

* update the confi

* upate files

* update baselines

* Create .npmrc

* Update .npmrc

* update conf file with node options

* update the wdio test files and config

* Update .npmrc

* update the node options

* update the code to use cjs

* Update package.json

* update the act conf

* Create .npmrc

* add node option in act test

* Update package.json

---------




* New Rule(s): **List rule IDs**
* New or modified help files: **List rule IDs**
* Rule bug: **List rule IDs**
* Extension UI bug
* Automated tool bug
* Other (Provide information)

### This PR is related to the following issue(s): 

- 

### Testing reference: 

- 

### I have conducted the following for this PR: 
- [ ] I validated this fix in my local env
- [ ] I provided details for testing
- [ ] I understand that the title of this PR will be used for the next release notes.


### Assets to aide review attached

- [ ] Links to design artifacts 
- [ ] Links to video walkthrough of user experience 

### Definition of Done

- [ ] Peer review complete
- [ ] Secondary review complete
- [ ] Staging deployment verified
